catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -454,6 +454,12 @@ private Object interpretFunction(Functions function, Sprite sprite) {
 				return doubleValueOfLeftChild == null ? defaultReturnValue : java.lang.Math.toDegrees(Math.acos(doubleValueOfLeftChild));
 			case ARCTAN:
 				return doubleValueOfLeftChild == null ? defaultReturnValue : java.lang.Math.toDegrees(Math.atan(doubleValueOfLeftChild));
+			case ARCTAN2:
+				return (doubleValueOfLeftChild == null || doubleValueOfRightChild == null) ? defaultReturnValue
```

Comment:
I know that all of the other formula evaluations are written just as compact as this one, but this is really really hard to understand. the thing is: if someone has to change this they're going to have a bad time (it might even be difficult for you to know what's going on here if you have to look at it again in the future).

so what I'm trying to say is: please convert this to clean code, i.e., something that can be understood more clearly from just looking at it. especially the nested ternary operators are a nightmare.

how about something like:
```
if (doubleValueOfLeftChild == null || doubleValueOfRightChild == null) {
    return defaultReturnValue;
}

if (doubleValueOfLeftChild == 0) && (doubleValueOfRightChild == 0) {
    return -180 + java.lang.Math.random() * 360;
} else {
    return java.lang.Math.toDegrees(java.lang.Math.atan2(doubleValueOfLeftChild, doubleValueOfRightChild)));
}
```

the whole formula parsing is complicated enough as it is (I can't even count the number of people who unsuccessfully try to find someone who knows what's going on here), so I'd really appreciate some development towards a more sefl-explaining imlementation.

Reply:
Changed it, just thought I didn't want to produce the only piece of code in that file that does not look as compact as the rest.

Reply:
sure, I suspected as much and it's a perfectly justified approach. it's just that in a project like this, code sometimes ages badly and many design decisions (such as using very compact code here) are just no longer reasonable. so I think if you change anything you don't necessarily have to follow all conventions all the time. don't get me wrong, conventions and coding standards are a good thing, but sometimes conventions are (no longer) reasonable.

**TL;DR:** I understand why you did it like this, thanks for changing it anyway.